app/src/main/java/com/cornellappdev/android/eatery/presenter/MainListPresenter.java

```diff
@@ -0,0 +1,94 @@
+package com.cornellappdev.android.eatery.presenter;
+
+import android.view.View;
+import android.widget.Button;
+
+import com.cornellappdev.android.eatery.MainListFragment;
+import com.cornellappdev.android.eatery.R;
+import com.cornellappdev.android.eatery.Repository;
+import com.cornellappdev.android.eatery.model.EateryBaseModel;
+import com.cornellappdev.android.eatery.model.enums.CampusArea;
+import com.cornellappdev.android.eatery.model.enums.PaymentMethod;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+public class MainListPresenter {
+
+	private ArrayList<EateryBaseModel> eateryList;
+	private ArrayList<EateryBaseModel> currentList;
+	private Set<Button> areaButtonsPressed;
+	private Set<Button> paymentButtonsPressed;
+	private HashSet<PaymentMethod> paymentSet;
+	private HashSet<CampusArea> areaSet;
+
+	public Repository rInstance = Repository.getInstance();
```

Comment:
`repositoryInstance`, should all of these really be public, I haven't seen a reference to one from a different class? (I could've just not been reading carefully enough)

Reply:
you're right, they should be private